packages/gatsby-theme-aio/gatsby-node.js

```diff
@@ -36,75 +36,82 @@ exports.createSchemaCustomization = ({ actions }) => {
       title: String
       path: String
     }
-    
+
     type Menu {
       title: String
       description: String
       path: String
     }
-    
+
     type TopPage {
       title: String
       path: String
       menu: [Menu]
     }
-    
+
     type Version {
       title: String
       path: String
       selected: Boolean
     }
-    
+
     type Home {
       title: String
       path: String
       hidden: Boolean
     }
-    
+
     type SiteSiteMetadata {
       home: Home
       pages: [TopPage]
       subPages: [SubPage]
       versions: [Version]
       docs: Link
     }
-    
+
     type SubPage {
       title: String
       path: String
       header: Boolean
       pages: [NestedSubPage1]
     }
-    
+
     type NestedSubPage1 {
       title: String
       path: String
       pages: [NestedSubPage2]
     }
-    
+
     type NestedSubPage2 {
       title: String
       path: String
       pages: [NestedSubPage3]
     }
-    
+
     type NestedSubPage3 {
       title: String
       path: String
       pages: [NestedSubPage4]
     }
-    
+
     type NestedSubPage4 {
       title: String
       path: String
       pages: [NestedSubPage5]
     }
-    
+
     type NestedSubPage5 {
       title: String
       path: String
       pages: [Link]
     }
+
+    type MdxFrontmatter {
```

Comment:
Yes, I added `keywords` as an explicit schema type [String] (string array) to prevent errors when there isn't a `keywords` present within any of the frontmatter for a site. That way Gatsby doesn't depend on its existence to make an inference about its type. But I did also added `title: String, description: String, contributors: [String]` one of which may already be defined for MdxFrontmatter by one of the plugins. I'll check and make sure, submit a fix that gets rid of the warning.

Reply:
> to prevent errors when there isn't a keywords present within any of the frontmatter for a site.

Pages without frontmatter didn't fail previously so do you mean with the new Algolia code especially `keywords`, we have to define the schema else it would break ? 
I'm just trying to understand what this code is fixing :)

Reply:
Committed a separate PR for this.